[Feat] 카테고리 생성 API 구현

BAEK0111 opened the pull request:
## #️⃣연관된 이슈
> #9 

## 📝작업 내용
> 카테고리 생성 API 구현 완료
> 스웨거 테스트 완료
> 데이터베이스 테스트 완료

## 🔎코드 설명
> DTO, Converter, Controller, Service, Repository 생성

## 💬고민사항 및 리뷰 요구사항 (Optional)
> 고민사항 및 의견 받고 싶은 부분 있으면 적어두기

## 비고 (Optional)
> 참고했던 링크 등 참고 사항을 적어주세요. 코드 리뷰하는 사람이 참고해야 하는 내용을 자유로운 형식으로 적을 수 있습니다.

BAEK0111 committed:
### 제목 - (영문기준) 50자 이내
# chore: #9 데이터 그립 연결

### 본문 - (영문 기준) 1줄 72자 이내
# 현재 datagrip에 연결 완료

BAEK0111 committed:
### 제목 - (영문기준) 50자 이내
# feat: #9 카테고리 생성 api 구현

### 본문 - (영문 기준) 1줄 72자 이내
# 카테고리 생성 api 구현 완료 && 스웨거, 데이터그립 테스트 완료

BAEK0111 committed:
Merge branch 'develop' of https://github.com/Ttatta-org/Ttatta_BE into feature/#9-create-category-api

Jinho622 commented on src/main/java/TtattaBackend/ttatta/domain/DiaryCategories.java:
```diff
@@ -32,6 +33,8 @@ public class DiaryCategories extends BaseEntity {
     @Column(columnDefinition = "VARCHAR(10) DEFAULT 'DEFAULT'")
     private CategoryColor color;
 
+    private LocalDateTime date;
```
DiaryCategories에 BaseEntity로 createdAt과 updatedAt이 들어가있어서 Entity가 생성되거나 변경될때마다 알아서 처리가 돼요. 그래서 date는 추가할 필요 없습니다. date는 삭제하고 다시 commit날려주세요.
그리고 관련 로직들에서 date를 createdAt으로 변경해주세요.

Jinho622 commented on src/main/java/TtattaBackend/ttatta/web/controller/DiaryCategoryController.java:
```diff
@@ -0,0 +1,24 @@
+public class DiaryCategoryController {
+    private final DiaryCategoryCommandService diaryCategoryCommandService;
+
+    @PostMapping("/")
```
테스트 유저 생성하는 api를 참고해서 @Operation 어노테이션을 사용해 swagger에 api에 대한 설명을 작성해주세요. 어떤 api인지, 요청을 보낼때 어떤 데이터를 넣어 보내주어야 하는지 써주시면 됩니다. 프론트 개발자분들이 api를 사용할 때 도움이 돼요!

BAEK0111 committed:
### 제목 - (영문기준) 50자 이내
# feat: #9 카테고리 생성 api 구현

### 본문 - (영문 기준) 1줄 72자 이내
# Controller에 @Operation 어노테이션을 사용하여 api 설명 작성

BAEK0111 committed:
### 제목 - (영문기준) 50자 이내
# fix: #9 카테고리 생성 api 구현

### 본문 - (영문 기준) 1줄 72자 이내
# date 필드 제거
# requestDTO에 userId 추가

BAEK0111 committed:
### 제목 - (영문기준) 50자 이내
# feat: #9 카테고리 생성 api 구현

### 본문 - (영문 기준) 1줄 72자 이내
# 양방향 매핑 편의 테이블 사용

Jinho622 commented on src/main/java/TtattaBackend/ttatta/converter/DiaryCategoryConverter.java:
```diff
@@ -0,0 +1,59 @@
+        return DiaryCategories.builder()
+                .name(request.getCategoryName())
+                .color(diaryCategoryColor)
+                .diariesList(new ArrayList<>())
```
setUser에서 연관관계를 처리해주고 있어서 .diariesList(new ArrayList<>())는 지워주셔도 될 것 같아요.

Jinho622 commented on src/main/java/TtattaBackend/ttatta/converter/DiaryCategoryConverter.java:
```diff
@@ -0,0 +1,59 @@
+                .categoryId(diaryCategory.getId())
+                .categoryColor(diaryCategory.getColor())
+                .categoryName(diaryCategory.getName())
+                .createdAt(LocalDateTime.now())
```
LocalDateTime.now()를 diaryCategory.getCreatedAt()으로 바꿔주세요.

BAEK0111 committed:
### 제목 - (영문기준) 50자 이내
# [커밋 타입]: [#이슈번호] [작업 내용]
# ex) feat: #3 유저 삭제 기능 구현
fix: #9 컨버터 수정

### 본문 - (영문 기준) 1줄 72자 이내
# (1)무엇을 (2)왜 (3) 어떻게 수정했는지
.diariesList(new ArrayList<>()) 삭제
.createdAt(LocalDateTime.now) -> .createdAt(diaryCategory.getCreatedAt())

### 꼬리말
# (선택)이슈 등 아래 작성

#   [ 커밋 타입 목록 ]
#   feat	:  기능 (새로운 기능)
#   fix		:  버그 (버그 수정)
#   refactor	:  리팩토링
#   style	:  스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs	:  문서 (문서 추가, 수정, 삭제)
#   test	:  테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore	:  기타 변경사항, 설정 관련 변경사항 (빌드 스크립트 수정, package.json 등)
